client/components/search/index.js

```diff
@@ -71,55 +78,111 @@ class Search extends Component {
 		}
 	}
 
+	renderTags() {
+		const { selected } = this.props;
+		return selected.length ? (
+			<div className="woocommerce-search__token-list">
+				{ selected.map( ( item, i ) => {
+					const screenReaderLabel = sprintf(
+						__( '%1$s (%2$s of %3$s)', 'wc-admin' ),
+						item.label,
+						i,
+						selected.length
+					);
+					return (
+						<Tag
+							key={ item.id }
+							id={ item.id }
+							label={ item.label }
+							remove={ this.removeResult }
+							removeLabel={ __( 'Remove product', 'wc-admin' ) }
+							screenReaderLabel={ screenReaderLabel }
+						/>
+					);
+				} ) }
+			</div>
+		) : null;
+	}
+
+	onFocus() {
+		this.setState( { isActive: true } );
+	}
+
+	onBlur() {
+		this.setState( { isActive: false } );
+	}
+
 	render() {
 		const autocompleter = this.getAutocompleter();
-		const { placeholder, selected } = this.props;
-		const { value = '' } = this.state;
+		const { placeholder, inlineTags, selected, instanceId } = this.props;
+		const { value = '', isActive } = this.state;
 		const aria = {
 			'aria-labelledby': this.props[ 'aria-labelledby' ],
 			'aria-label': this.props[ 'aria-label' ],
 		};
 		return (
 			<div className="woocommerce-search">
+				<label htmlFor={ `search-input-${ instanceId }` } className="screen-reader-text">
+					{ placeholder || __( 'Add item', 'wc-admin' ) }
```

Comment:
We're already using `aria-label` or `aria-labelledby` to fill this purpose, we don't actually need to add a `label` here (especially an invisible label).

Reply:
Oh right of course. Those are passed in via props. How do you feel about making them `required`?

Reply:
Well, you wouldn't have both– `aria-label` would just be a string, and `aria-labelledby` is an element ID. If you want to have a required prop, we could change it to accept a `label` prop, which would be a string, and we can either pass it to the input's `aria-label` or this `<label>` element. We'd have to change how the `SearchFilter` works, though (that uses `ariaLabelledby`, which is actually not right).

Reply:
>We'd have to change how the SearchFilter works, though (that uses ariaLabelledby, which is actually not right).

This reminds me, I've changed this in https://github.com/woocommerce/wc-admin/pull/457 to use `aria-label` as suggested